src/bootstrap/tool.rs

```diff
@@ -178,12 +184,29 @@ impl Step for ToolBuild {
                       typically means that something was recompiled because \
                       a transitive dependency has different features activated \
                       than in a previous build:\n");
+            println!("the following dependencies are duplicated although they \
+                      have the same features enabled:");
+            for (id, cur, prev) in &duplicates {
+                println!("  {}", id);
+                if cur.2 == prev.2 {
+                    // same features
+                    println!("    `{}` ({:?})\n    `{}` ({:?})", cur.0, cur.1, prev.0, prev.1);
+                }
+            }
+            println!("the following dependencies have different features:");
             for (id, cur, prev) in duplicates {
                 println!("  {}", id);
-                println!("    `{}` enabled features {:?} at {:?}",
-                         cur.0, cur.2, cur.1);
-                println!("    `{}` enabled features {:?} at {:?}",
-                         prev.0, prev.2, prev.1);
+                if cur.2 == prev.2 {
+                    continue;
```

Comment:
You could use [`duplicates.drain_filter(...)`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.drain_filter) to avoid the double-checking.

Reply:
^^ I considered that and then was like oh it's just during failure and copy paste is so convenient. I'll do it anyway.